### PR TITLE
304 passthrough for AzureBlobCache.streamCachedImage

### PR DESCRIPTION
**Pull request by JasonElkin**
Fix for #428 

Maps HTTP headers between browser and Azure CDN to enable automatic handling of 304s

### CODE REVIEW

**Comment by JasonElkin** [comment]
While testing I noticed that the AzureBlobCache makes a lot of unnecessary (and taxing) calls to the Azure Blob Storage API.

I was going to raise it as a separate issue but I accidentally added the fix to this Pull Request.

Anyways, the changes save ≈500ms (at least in my test environment).

File: src/ImageProcessor.Web.Plugins.AzureBlobCache/AzureBlobCache.cs [hunk 1]
```diff
@@ -378,24 +392,58 @@ public override void RewritePath(HttpContext context)
...
+                    {
+                        HttpWebResponse response;
+                        try
                         {
-                            using (HttpWebResponse response = (HttpWebResponse)request.GetResponse())
+                            response = (HttpWebResponse)request.GetResponse();
+                        }
+                        catch (WebException ex)
+                        {
+                            // A 304 is not an error
```
Comment (JimBobSquarePants) [inline src/ImageProcessor.Web.Plugins.AzureBlobCache/AzureBlobCache.cs]: Can't believe I missed that. I need to check elsewhere in my codebase now for similar!

**Comment by JimBobSquarePants** [comment]
Great stuff! Thanks for this :+1: The added performance benefits are massive also so very pleased with the work. 

Very much appreciated!

**Comment by alindgren** [comment]
@JasonElkin awesome! I noticed the calls too - this should help us out a lot!

**Pull request merged**